### ISSUE TITLE
[no-shadow] Add `ignoreOnInitialization` option to match the ESLint rule

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#4601]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-shadow": ["error", { "ignoreOnInitialization": true }],
  }
}
```

```TS
// n/a
```

**Expected Result**

No errors.

**Actual Result**

```
Configuration for rule "@typescript-eslint/no-shadow" is invalid:
Value {"ignoreOnInitialization":true} should NOT have additional properties.
```

**Additional Info**

https://eslint.org/docs/rules/no-shadow#ignoreoninitialization

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.12.1` |
| `@typescript-eslint/parser`        | `5.12.1` |
| `TypeScript`                       | `4.5.5` |
| `ESLint`                           | `8.10.0` |
| `node`                             | `16.14.0` |
